crates/rattler_shell/src/shell/mod.rs

```diff
@@ -421,7 +422,17 @@ impl Shell for PowerShell {
     }
 
     fn executable(&self) -> &str {
-        self.executable_path.as_deref().unwrap_or("pwsh")
+        if let Some(executable_path) = self.executable_path.as_deref() {
+            executable_path
+        } else {
+            // check if `powershell` is available, if not, use `pwsh`
```

Comment:
Can we do this as part of the construction?

Reply:
I am doing this now in `default`.